src/components/MapSelect/index.js

```diff
@@ -0,0 +1,195 @@
+    return iconSet.selected;
+  }
+  return iconSet.default;
+};
+
+class MapSelect extends React.Component {
+  /**
+   * Multiple features select map
+   * Required geojson endpoint supporting bbox parameter
+   */
+  componentDidMount() {
+    const {
+      latlng: centerLatLng,
+      geojsonUrl,
+      onSelectionChange,
+      iconMapping,
+      iconField,
+      idField,
+      zoomMin,
+      value
+    } = this.props;
+
+    this.map = amaps.createMap({
+      center: {
+        latitude: centerLatLng.latitude,
+        longitude: centerLatLng.longitude
+      },
+      layer: 'standaard',
+      target: 'mapdiv',
+      marker: false,
+      search: false,
+      zoom: ZOOM_INIT
+    });
+
+    const errorControl = new ErrorControl({
+      position: 'topleft',
+      message: 'Oops, de objecten kunnen niet worden getoond. Probeer het later nog eens.',
+    });
+
+    const selection = new MaxSelection(SELECTION_MAX_COUNT, value);
+    onSelectionChange(selection);
+    this.selection = selection;
+
+    // istanbul ignore next
+    const fetchRequest = (bbox_str) => request(`${geojsonUrl}&bbox=${bbox_str}`)
+      .catch((e) => {
+        console.error('Error loading feature geojson', e); // eslint-disable-line no-console
+        errorControl.show();
+      });
+
+    this.featuresLayer = BboxGeojsonLayer({
+      fetchRequest,
+    }, {
+      zoomMin,
+
+      pointToLayer(feature, latlng) {
+        // istanbul ignore next
+        return L.marker(latlng, {
+          icon: getIcon(iconMapping, feature.properties[iconField], selection.has(feature.properties[idField]))
+        });
+      },
+
+      onEachFeature(feature, layer) {
+        // istanbul ignore next
+        layer.on({
+          click: (e) => {
+            const _layer = e.target;
+            const id = _layer.feature.properties[idField];
+            selection.toggle(id);
+            onSelectionChange(selection);
+          }
+        });
+      }
+    });
+    this.featuresLayer.addTo(this.map);
+
+    const zoomMessageControl = new ZoomMessageControl({
+      position: 'topleft',
+      zoomMin
+    });
+    zoomMessageControl.addTo(this.map);
+
+    const legendControl = new LegendControl({
+      position: 'topright',
+      zoomMin,
+      elements: this.props.legend
+    });
+    legendControl.addTo(this.map);
+
+    const div = L.DomUtil.create('div', 'loading-control');
+    div.innerText = 'Bezig met laden...';
+
+    const loadingControl = new LoadingControl({
+      position: 'topleft',
+      element: div
+    });
+    loadingControl.addTo(this.map);
+
+    errorControl.addTo(this.map);
+  }
+
+  componentDidUpdate(prevProps) {
+    const lat = this.props.latlng.latitude;
+    const lng = this.props.latlng.longitude;
+    if (lat !== prevProps.latlng.latitude || lng !== prevProps.latlng.longitude) {
+      this.map.panTo({
+        lat,
+        lng
+      });
+    }
+
+    const value = this.props.value;
+    if (value !== prevProps.value) {
```

Comment:
om een array te checken voor verandering heb je toch isEqual nodig?

Reply:
Niet als de onderliggende referentie wijzigd. Maar ik pas het aan. isEqual is wel passender hier